pkg/reconciler/buildrun/buildrun.go

```diff
@@ -147,16 +147,25 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			if err != nil {
 				return reconcile.Result{}, err
 			}
-
 			ctxlog.Info(ctx, "creating TaskRun from BuildRun", namespace, request.Namespace, name, generatedTaskRun.GenerateName, "BuildRun", buildRun.Name)
 			if err = r.client.Create(ctx, generatedTaskRun); err != nil {
 				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+				serviceAccountName := &generatedTaskRun.Spec.ServiceAccountName
```

Comment:
I´m assuming this is a way to ensure we populate the `buildRun.Status.ServiceAccountName` in case the creation of the TaskRun failed. I think we do not need this, because if the creation of the TaskRun failed, we will reconcile again and eventually reach this [line](https://github.com/shipwright-io/build/pull/608/files?file-filters%5B%5D=.go#diff-92a124fc33b3ec7d38392c0b313144f552c7987f35bb1aad07ea355020ad32e3R167)